lib/Site.js

```diff
@@ -573,6 +577,20 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
   });
 };
 
+/**
+ * Recollects the user defined variables map in the site/subsites
+ * if there is a change in the variables file
+ * @param filePaths array of paths corresponding to files that have changed
+ */
+Site.prototype.recollectUserDefinedVariablesMapIfNeeded = function (filePaths) {
```

Comment:
- Let's go with `collect` rather than `recollect` (similar for the jsdoc).
  Just as it's not **re**[scrollIntoViewIfNeeded](
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoViewIfNeeded) or **re**[layoutIfNeeded](
https://developer.apple.com/documentation/uikit/uiview/1622507-layoutifneeded).

- I should have been more specific when I said "Place it just under the function that it wraps".
  I meant `collectUserDefinedVariablesMap`.